sljit_src/sljitNativeRISCV_common.c

```diff
@@ -1043,10 +1073,24 @@ static sljit_s32 getput_arg(struct sljit_compiler *compiler, sljit_s32 flags, sl
 		next_arg = 0;
 		next_argw = 0;
 	}
-
 	if (SLJIT_UNLIKELY(arg & OFFS_REG_MASK)) {
 		argw &= 0x3;
-
+#if defined __riscv_zba
+		switch (argw) {
+			case 0: 
+				FAIL_IF(push_inst(compiler, ADD | RD(tmp_r) | RS1(OFFS_REG(arg)) | RS2(base)));
+				break;
+			case 1:
+				FAIL_IF(push_inst(compiler, SH1ADD | RD(tmp_r) | RS1(OFFS_REG(arg)) | RS2(base)));
+				break;
+			case 2:
+				FAIL_IF(push_inst(compiler, SH2ADD | RD(tmp_r) | RS1(OFFS_REG(arg)) | RS2(base)));
+				break;
+			case 3:
+				FAIL_IF(push_inst(compiler, SH3ADD | RD(tmp_r) | RS1(OFFS_REG(arg)) | RS2(base)));
```

Comment:
These are essentially the same except the opcode changes. We could put the instruction into a local variable (`sljit_ins ins`) and call push_inst once.